sdcm/nemesis.py

```diff
@@ -2324,7 +2324,19 @@ def remove_node():
             # the entire nemesis.
 
             exit_status = remove_node()
-            assert exit_status == 0, "nodetool removenode command exited with status {}".format(exit_status)
+
+            # if remove node command failed by any reason,
+            # we will remove the terminated node from
+            # dead_nodes_list, so the health validator terminate the job
+            if exit_status != 0:
+                self.log.error(f"nodetool removenode command exited with status {exit_status}")
```

Comment:
Failure of `nodetool removenode` is not expected.
I suggest to report Error event instead of log error. Log error will be ignored.

Reply:
an error event will surely have more visibility and in the end, if that will be the only error, the test will be marked as failed, but in other hand, IIUC from the issue that motivated this job, removing it from the `dead_nodes_list` will make the ClusterHealthCheck to fail, and then it will raise an event and will probably stop the test...

Reply:
I agree that this node should be removed from the `dead_nodes_list` to prevent critical error. But we should know that `removenode` failed and check why. If log.error will be printed only (not error event), we will miss it

Reply:
@juliayakovlev do yo think the critical nodetool event will not be enough for that?

Reply:
You can publish critical nodetool event if you want that the test will be stopped. Now when nodetool command fails with exception, ERROR or WARNING will be published. If no exception but returned status is not 0 (as you checked here), you need to handle it as you need in the test. You won't receive CRITICAL nodetool event

Reply:
I think there was already similar PR to that by Yulia and we decided not to do it.
I don't want the test to fail completely just because the remove node failed.

Reply:
@roydahan what we going to do then? If remove node failed ( it is already terminated and doesn't exists) but it stay in cluster with status DN. Current implementation will just report error message, the Nemesis will faile by assert that node was not properly removed from cluster, and the HealthValidator will fail the test because one of node have statud DN.

Reply:
Ok, now I understand why the ClusterHealthValidator wouldn't catch it.